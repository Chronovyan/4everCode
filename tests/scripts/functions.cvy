// Test of function definitions and calls in Chronovyan

// Define a simple function that prints a message
function greet(name) {
    print("Hello, " + name + "!");
}

// Define a function that calculates the factorial of a number
function factorial(n) {
    if (n <= 1) {
        return 1;
    } else {
        return n * factorial(n - 1);
    }
}

// Define a function that checks if a number is prime
function isPrime(num) {
    if (num <= 1) {
        return false;
    }
    if (num <= 3) {
        return true;
    }
    if (num % 2 == 0 || num % 3 == 0) {
        return false;
    }
    
    var i = 5;
    while (i * i <= num) {
        if (num % i == 0 || num % (i + 2) == 0) {
            return false;
        }
        i = i + 6;
    }
    return true;
}

// Test function calls
print("Testing function calls:");

// Test greet function
greet("Chronovyan");

// Test factorial function
print("Factorial of 5 is: " + factorial(5));

// Test isPrime function
var testNumbers = [2, 3, 4, 5, 6, 7, 11, 12, 13];
for (var i = 0; i < 9; i = i + 1) {
    var num = testNumbers[i];
    if (isPrime(num)) {
        print(num + " is prime");
    } else {
        print(num + " is not prime");
    }
} 