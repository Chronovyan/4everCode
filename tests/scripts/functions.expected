Testing function calls:
Hello, Chronovyan!
Factorial of 5 is: 120
2 is prime
3 is prime
4 is not prime
5 is prime
6 is not prime
7 is prime
11 is prime
12 is not prime
13 is prime 